site/cds_rdm/components.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 CERN.
+#
+# CDS-RDM is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS service component for subjects validation."""
+
+from invenio_drafts_resources.services.records.components import ServiceComponent
+from marshmallow import ValidationError
+
+
+class SubjectsValidationComponent(ServiceComponent):
+    """Service component for subjects validation."""
+
+    def _validate_subject_changes(self, identity, updated_data, original_data):
+        """Validate that the subject changes are allowed."""
+        if identity.user.has_role("admin"):
```

Comment:
This should be good enough for us, however in general a user could be attached to a role or an action, not always a role.
We normally check if the `identity` provides a Need, which is the expanded `superuser_access` action.